applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -51,20 +43,33 @@
      * @param serviceEndpoint URL for the Application configuration service.
      * @param pipeline HttpPipeline that the HTTP requests and responses flow through.
      */
-    private ConfigurationClient(URL serviceEndpoint, HttpPipeline pipeline) {
+    ConfigurationClient(URL serviceEndpoint, HttpPipeline pipeline) {
         super(pipeline);
 
-        this.service = RestProxy.create(ApplicationConfigService.class, this);
+        this.service = RestProxy.create(ConfigurationService.class, this);
         this.serviceEndpoint = serviceEndpoint.toString();
     }
 
     /**
      * Creates a builder that can configure options for the ConfigurationClient before creating an instance of it.
      *
-     * @return A new Builder to create a ConfigurationClient from.
+     * @return A new ConfigurationClientBuilder to create a ConfigurationClient from.
+     */
+    public static ConfigurationClientBuilder builder() {
+        return new ConfigurationClientBuilder();
+    }
+
+    /**
+     * Adds a configuration value in the service if that key does not exist.
+     *
+     * @param key The key for the configuration setting to add.
+     * @param value The value associated with this configuration setting key.
+     * @return ConfigurationSetting that was created.
+     * @throws IllegalArgumentException If {@code key} or {@code value} are {@code null} or an empty string.
```

Comment:
I'm guessing you're going to follow-up with the need to reduce client-side validation?